lucille-core/src/main/java/com/kmwllc/lucille/connector/FileConnector.java

```diff
@@ -158,42 +158,63 @@ public FileConnector(Config config) throws ConnectorException {
             S3_PARENT_SPEC
         ));
 
-    this.pathToStorage = config.getString("pathToStorage");
     this.fileOptions = config.hasPath("fileOptions") ? config.getConfig("fileOptions") : ConfigFactory.empty();
     this.filterOptions = config.hasPath("filterOptions") ? config.getConfig("filterOptions") : ConfigFactory.empty();
+    this.storageClientMap = StorageClient.createClients(config);
 
-    try {
-      this.storageURI = new URI(pathToStorage);
-      log.debug("using path {} with scheme {}", pathToStorage, storageURI.getScheme());
-    } catch (URISyntaxException e) {
-      throw new ConnectorException("Invalid path to storage: " + pathToStorage, e);
+    List<String> pathsToStorage = config.getStringList("pathsToStorage");
+    this.storageURIs = new ArrayList<>();
+
+    for (String path : pathsToStorage) {
+      try {
+        URI newStorageURI = new URI(path);
+        storageURIs.add(newStorageURI);
+        log.debug("FileConnector to use path {} with scheme {}", path, newStorageURI.getScheme());
+      } catch (URISyntaxException e) {
+        throw new ConnectorException("Invalid path to storage: " + path, e);
+      }
     }
 
-    if (CLOUD_STORAGE_CLIENT_KEYS.stream().filter(config::hasPath).count() > 1) {
-      log.warn("Config for FileConnector contains options for more than one cloud provider.");
+    // Cannot specify multiple storage paths and a moveTo of some kind
+    if (storageURIs.size() > 1 && (config.hasPath("fileOptions.moveToAfterProcessing") || config.hasPath("fileOptions.moveToErrorFolder"))) {
+      throw new IllegalArgumentException("FileConnector does not support multiple pathsToStorage and moveToAfterProcessing / moveToErrorFolder. Create individual FileConnectors.");
     }
   }
 
   @Override
   public void execute(Publisher publisher) throws ConnectorException {
     try {
-      storageClient = StorageClient.create(storageURI, config);
-    } catch (Exception e) {
-      throw new ConnectorException("Error occurred while creating storage client.", e);
+      for (StorageClient client : storageClientMap.values()) {
```

Comment:
should this be included in the try/finally below?
as it stands now, we might init some clients, then encounter an exception on a subsequent client and never shutdown any of them